### ISSUE TITLE
how to use current() in identity()

### ISSUE THREAD

**Issue by scuben** [nelmio/alice#374]
I found the documentation about identity and its use of variables in it. I also found several issues regarding using current() in identity(). Still I did not manage to get it working.

As far as I could read in the issues it has been implemented. So... Can anyone provide an example? As a bonus you could in that case directly add a PR explaining it in the docs.

**Comment by theofidry** [comment]
Hello,

`identity()` is a simple "faker" function, not provided by faker natively, but by a [Faker provider](https://github.com/nelmio/alice/blob/3bf8e6f9cabb4f7d06d0c0d8f715ceb4b41d19c9/doc/customizing-data-generation.md#customizing-data-generation) still: [IdentityProvider](https://github.com/nelmio/alice/blob/3bf8e6f9cabb4f7d06d0c0d8f715ceb4b41d19c9/src/Nelmio/Alice/Instances/Processor/Providers/IdentityProvider.php).

You have a piece of doc [here](https://github.com/nelmio/alice/blob/3bf8e6f9cabb4f7d06d0c0d8f715ceb4b41d19c9/doc/customizing-data-generation.md#default-providers), but I agree it could be better. However here's some examples:

```yaml
Nelmio\Entity\User:
  user1:
    username: 'testuser'
    fullname: '<identity($username)>'
  user2:
    username: 'test_user'
    fullname: '<identity(str_replace("_", " ", $username))>'
```

`<current()>` is also explained [here](https://github.com/nelmio/alice/blob/3bf8e6f9cabb4f7d06d0c0d8f715ceb4b41d19c9/doc/relations-handling.md#references). Same I think it could be better so any PR to help is welcomed :)

In short, `<current()`> can be used when dealing with a fixture collection such as `user_{1..10}` or `user_{alice, bob}` and returns the "current" value of the collection:

```yaml
Nelmio\Entity\User:
  user_{1..10}: # creates 10 users "user_1", ..., "user_10"
    username: '<current()>' # will be "1", ..., "10"
  user_{alice, bob}: # creates two users "user_alice" and "user_bob"
    username: '<current()>' # respectively "alice" and then "bob"
```

If my memory serves me right, it throws an exception if you try to use `<current()>` outside of that context (needs to be checked).

Best,

**Comment by scuben** [comment]
Hi @theofidry 

Thanks for your response. I know both features `<identity()>` and `<current()>`. But what I still dont know about how to use both at the same time. I try to have `<current()>` subtracted with 2.

What I want to achieve is this (which is not working):
```yaml
Acme\Entity\User:
  user_ {3..10}:
    anything: <identity(<current()> - 2)>
```

Is there a syntax which will give me 1..8 or is this not possible at all?

**Comment by theofidry** [comment]
Ah, as mentioned in #360 the correct syntax would be `'<identity(current() - 2)>'` but I couldn't test if it was working or not... :/

**Comment by scuben** [comment]
I just gave it a try but did not work:

```
[Symfony\Component\Debug\Exception\ContextErrorException]
Warning: current() expects exactly 1 parameter, 0 given
```

**Comment by theofidry** [comment]
Indeed, I'm not sure why I couldn't find much after a bit of debugging. I'm honestly not sure that this will be fixed in v2. In the meantime what you can do is having a provider to do the calculation:

```yaml
Acme\Entity\User:
  user_ {3..10}:
    anything: '<anything(current())>'
```

**Comment by scuben** [comment]
Not sure if that works as `<identity(current())>` did not. But I'll give it a try when i come across this issue again.

**Comment by theofidry** [comment]
Erf, I don't know then. I can only make sure that this will be fixed in v3 for now. If I find a fix on the way I'll make sure to include it.

**Comment by theofidry** [comment]
Closing in favour of https://github.com/nelmio/alice/issues/360
